File: typescript-examples/setup-hooks/utils/typesAndSchemas.ts
import { z } from "zod";

// Params schema for demo-hook API
export const demoHookParamsSchema = z.object({
  message: z.string().optional(),
});

// Data from hook schema
export const dataFromHookSchema = z.object({
  cdpUrl: z.string(),
  apiName: z.string(),
  apiParameters: z.any(),
  executionStartTime: z.string(),
  config: z.any(),
  userAgent: z.string(),
});

// Hook demo result schema
export const hookDemoResultSchema = z.object({
  message: z.string(),
  dataFromHook: dataFromHookSchema,
  executionTime: z.string(),
  pageTitle: z.string(),
});

// Type exports inferred from schemas
export type DemoHookParams = z.infer<typeof demoHookParamsSchema>;
export type DataFromHook = z.infer<typeof dataFromHookSchema>;
export type HookDemoResult = z.infer<typeof hookDemoResultSchema>;

File: python-examples/setup-hooks/utils/types_and_schemas.py
from typing import Any

from pydantic import BaseModel


class DemoHookParams(BaseModel):
    """Parameters for demo-hook API"""

    message: str | None = None


class DataFromHook(BaseModel):
    """Data retrieved from setup hook"""

    cdp_url: str
    api_name: str
    api_parameters: Any  # Can be dict, string, or any type
    execution_start_time: str
    config: Any  # Can be any type
    user_agent: str


class HookDemoResult(BaseModel):
    """Result returned from demo-hook API"""

    message: str
    data_from_hook: DataFromHook
    execution_time: str
    page_title: str
